src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -459,6 +428,54 @@ public void stop() {
         }
     }
 
+    private Mqtt5Client createCrtClient() throws MessageClientException {
+        boolean isSSL = "ssl".equalsIgnoreCase(brokerUri.getScheme());
+
+        long port = brokerUri.getPort();
+        if (port < 0) {
+            port = isSSL ? DEFAULT_SSL_MQTT_PORT : DEFAULT_TCP_MQTT_PORT;
+        }
+
+        Mqtt5ClientOptions.Mqtt5ClientOptionsBuilder builder =
+                new Mqtt5ClientOptions.Mqtt5ClientOptionsBuilder(brokerUri.getHost(), port)
+                        .withLifecycleEvents(connectionEventCallback)
+                        .withPublishEvents(publishEventsCallback)
+                        .withSessionBehavior(Mqtt5ClientOptions.ClientSessionBehavior.REJOIN_POST_SUCCESS)
+                        .withOfflineQueueBehavior(Mqtt5ClientOptions.ClientOfflineQueueBehavior.FAIL_ALL_ON_DISCONNECT)
+                        .withConnectOptions(new ConnectPacket.ConnectPacketBuilder()
+                                .withRequestProblemInformation(true)
+                                .withClientId(clientId).build())
+                        // TODO configurable?
+                        .withMaxReconnectDelayMs(Duration.ofMillis(MAX_RECONNECT_DELAY_MS).toMillis())
+                        .withMinReconnectDelayMs(Duration.ofMillis(MIN_RECCONECT_DELAY_MS).toMillis());
+
+        if (isSSL) {
+            mqttClientKeyStore.listenToCAUpdates(this::reset);
```

Comment:
store `this::reset` in a final field, otherwise you're going to register reset multiple times.